events/permissions.py

```diff
@@ -185,26 +230,26 @@ def has_object_permission(self, request: Request, view, obj):
         if request.method in permissions.SAFE_METHODS:
             return True
 
-        if (
-            settings.WEB_STORE_INTEGRATION_ENABLED
-            and request.method in ("PUT", "PATCH")
-            and (
-                request.data.get("web_store_merchants")
-                or request.data.get("web_store_accounts")
-            )
-        ):
-            data_keys = request.data.keys()
-
-            if len(data_keys) > 1:
-                return (
-                    request.user.is_superuser
-                    or request.user.is_financial_admin_of(obj)
-                    and utils.organization_can_be_edited_by(obj, request.user)
-                )
-            else:
-                return request.user.is_superuser or request.user.is_financial_admin_of(
-                    obj
-                )
+        if request.method in ("PUT", "PATCH"):
+            if self._is_editing_organization_users(request.data, obj):
+                # User must a superuser to add or remove an existing organization's users.
+                return request.user.is_authenticated and request.user.is_superuser
+            elif self._is_editing_web_store_merchants_or_accounts(request.data):
+                # User must be a superuser or a financial admin with organization admin rights to
+                # edit an existing organization's web store merchants or accounts.
+                data_keys = request.data.keys()
+
+                if len(data_keys) > 1:
+                    return (
+                        request.user.is_superuser
+                        or request.user.is_financial_admin_of(obj)
+                        and utils.organization_can_be_edited_by(obj, request.user)
```

Comment:
clarifying to myself; so if user is modifying both; the merchants and accounts, the user needs to have organization admin rights AND the financial admin rights?
But when modifying only other (merch or account) then financial admin is sufficient?

Reply:
The logic here should go like:

- if editing merchants and/or accounts with other organization data given in the request, the user must be either a superuser or a financial admin with also organization editing (org admin) rights
- if editing only merchants and/or accounts (without other organization data), the user must be either a superuser or a financial admin

But actually, this check doesn't appear to work properly because if the user gives both merchants and accounts, then `if len(data_keys) > 1` will be `True` and the organization admin rights are required unnecessarily :thinking: 

Fix needed here, BRB.

Reply:
Now fixed